pallets/api/src/fungibles/tests.rs

```diff
@@ -9,165 +9,212 @@ use frame_support::{
 
 const ASSET: u32 = 42;
 
+type Event = crate::fungibles::Event<Test>;
+
 #[test]
 fn transfer_works() {
 	new_test_ext().execute_with(|| {
-		let amount: Balance = 100 * UNIT;
-		create_asset_and_mint_to(ALICE, ASSET, ALICE, amount);
-		let balance_before_transfer = Assets::balance(ASSET, &BOB);
-		assert_ok!(Fungibles::transfer(signed(ALICE), ASSET, BOB, amount / 2));
-		let balance_after_transfer = Assets::balance(ASSET, &BOB);
-		assert_eq!(balance_after_transfer, balance_before_transfer + amount / 2);
+		let value: Balance = 100 * UNIT;
+		let id = ASSET;
+		let from = Some(ALICE);
+		let to = Some(BOB);
+
+		create_asset_and_mint_to(ALICE, id, ALICE, value * 2);
+		let balance_before_transfer = Assets::balance(id, &BOB);
+		assert_ok!(Fungibles::transfer(signed(ALICE), id, BOB, value));
+		let balance_after_transfer = Assets::balance(id, &BOB);
+		assert_eq!(balance_after_transfer, balance_before_transfer + value);
+		System::assert_last_event(Event::Transfer { id, from, to, value }.into());
```

Comment:
Minor, but could just import the `System::assert_last_event` function to avoid the `System::` prefix in all the tests.

Reply:
Because it is an associated function of the Pallet struct, rather than a free-standing function at the crate level, you cannot directly import it using frame_system::assert_last_event. Instead, it must be accessed through the Pallet struct.

The mock::System::assert_last_event import path assumes that System (which is an alias for frame_system::Pallet) has been defined in a mock module and that the assert_last_event function is accessible directly from this alias. However, the assert_last_event function is tied to the Pallet implementation for the frame_system module and is not a standalone function that can be imported directly like this.